source/Calamari.Tests/Fixtures/Integration/Packages/PackageExtractorFixture.cs

```diff
@@ -47,6 +47,27 @@ public void ExtractPumpsFilesToFilesystem(Type extractorType, string extension,
             Assert.AreEqual(9, filesExtracted, "Mismatch in the number of files extracted"); //If you edit the nupkg file with Nuget Package Explorer it will add a _._ file to EmptyFolder and you'll get 5 here.
             Assert.AreEqual("Im in a package!", text.TrimEnd('\n'), "The contents of the extractd file do not match the expected value");
         }
+        
+        
+        [Test]
+        public void CanExtractZipFileContainingSpecialCharacters()
+        {
+            var fileName = GetFixtureResource("Samples", string.Format("utf8.Filenames-1.0.0.zip"));
+
+            Type extractorType = typeof(ZipPackageExtractor);
+            
+            var extractor = (IPackageExtractor) Activator.CreateInstance(extractorType, ConsoleLog.Instance);
```

Comment:
Is there any reason why the extractor can't be instantiated directly with `new`? That way you don't need to have the additional public constructor just for this test.

Reply:
way better solution, 🙏